x-pack/filebeat/input/httpjson/config_okta_auth.go

```diff
@@ -165,15 +199,23 @@ func generateOktaJWTPEM(pemdata string, cnf *oauth2.Config) (string, error) {
 	return signJWT(cnf, key)
 }
 
-func pemPKCS8PrivateKey(pemdata []byte) (any, error) {
+func pemPKCS8PrivateKey(pemdata []byte) (crypto.Signer, error) {
 	blk, rest := pem.Decode(pemdata)
 	if rest := bytes.TrimSpace(rest); len(rest) != 0 {
 		return nil, fmt.Errorf("PEM text has trailing data: %d bytes", len(rest))
 	}
 	if blk == nil {
 		return nil, errors.New("no PEM data")
 	}
-	return x509.ParsePKCS8PrivateKey(blk.Bytes)
+	key, err := x509.ParsePKCS8PrivateKey(blk.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	signer, ok := key.(crypto.Signer)
+	if !ok {
+		return nil, fmt.Errorf("key is not a signer: %T", key)
```

Comment:
I think we could improve on the clarity of this error.

```suggestion
		return nil, fmt.Errorf("'key' does not implement the Signer interface: type %T", key)
```

Reply:
The errors will be presented to users, the message currently reflects that.

Reply:
If I troubleshoot a support case and I saw "key is not a signer: WhateverType" that would not tell me anything. Let's give more details, so other engineers can also understand what's going on.

Reply:
If I were trouble-shooting this, I'd `rg "key is not a signer"` and find the location of the error origin. In reality, this error will never happen since all private keys satisfy `crypto.Signer`.

Reply:
Up to you, I don't have a strong opinion here.